updates for building pi3hat_tools for python3.11

JGSuw opened the pull request:
My implementation of support for python3.11 targets to pi3hat_tools, with a summary of changes listed in [this issue](https://github.com/mjbots/pi3hat/issues/16).

I have successfully built these targets on Ubuntu 22.04 using Python 3.11.7, including the pi3hat firmware and wheels for pi3/pi4 targets and python versions 3.7, 3.9, 3.10, and 3.11.

JGSuw commented on make_release.py:
I mistakenly left out the "dirty" checks in make_release.py, which I had removed to speed up testing of building the python wheels (building the firmware .elf was consuming a lot of time).

JGSuw replied:
Also, added --verbose_failures to the build command for pi3hat_tools, which was very useful in tracking down issues with the build configuration, but may be undesirable in the main repository.

jpieper commented:
To save me some time, how did you get python3.11 to be used on your Ubuntu 22.04 system?  I have it installed, but am not sure how to set it as the default so the build script uses it.

(to be honest, I had actually done this work back when I updated bazel_deps, but didn't bother figuring out how to get it to run as default then, hoping that 24.04 came out before anyone needed it)

JGSuw commented:
I added deadsnakes ppa with apt-add-repository, he runs nightly builds of new python versions and has builds for all(most?) of python3 versions

On Fri, Dec 15, 2023 at 4:37 AM, jpieper ***@***.***(mailto:On Fri, Dec 15, 2023 at 4:37 AM, jpieper <<a href=)> wrote:

> To save me some time, how did you get python3.11 to be used on your Ubuntu 22.04 system? I have it installed, but am not sure how to set it as the default so the build script uses it.
>
> (to be honest, I had actually done this work back when I updated bazel_deps, but didn't bother figuring out how to get it to run as default then, hoping that 24.04 came out before anyone needed it)
>
> —
> Reply to this email directly, [view it on GitHub](https://urldefense.com/v3/__https://github.com/mjbots/pi3hat/pull/17*issuecomment-1857817509__;Iw!!K-Hz7m0Vt54!jCtW8IPyUSzprNqWBUXrCEpED6e5eODfFDzJpSJubKqF0FjxgQ49StIzw9GAq7n0bOstchT8yFuJBoNHvzVDOSW-I4U$), or [unsubscribe](https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/ABF62HJMH5N52DQRK4N5SK3YJQ77LAVCNFSM6AAAAABAVPZGPKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNJXHAYTONJQHE__;!!K-Hz7m0Vt54!jCtW8IPyUSzprNqWBUXrCEpED6e5eODfFDzJpSJubKqF0FjxgQ49StIzw9GAq7n0bOstchT8yFuJBoNHvzVDYZpQWLU$).
> You are receiving this because you authored the thread.Message ID: ***@***.***>

JGSuw commented:
sudo add-apt-repository ppa:deadsnakes/ppa

On Fri, Dec 15, 2023 at 9:51 AM, Joseph G Sullivan ***@***.***(mailto:On Fri, Dec 15, 2023 at 9:51 AM, Joseph G Sullivan <<a href=)> wrote:

> I added deadsnakes ppa with apt-add-repository, he runs nightly builds of new python versions and has builds for all(most?) of python3 versions
>
> On Fri, Dec 15, 2023 at 4:37 AM, jpieper ***@***.***(mailto:On Fri, Dec 15, 2023 at 4:37 AM, jpieper <<a href=)> wrote:
>
>> To save me some time, how did you get python3.11 to be used on your Ubuntu 22.04 system? I have it installed, but am not sure how to set it as the default so the build script uses it.
>>
>> (to be honest, I had actually done this work back when I updated bazel_deps, but didn't bother figuring out how to get it to run as default then, hoping that 24.04 came out before anyone needed it)
>>
>> —
>> Reply to this email directly, [view it on GitHub](https://urldefense.com/v3/__https://github.com/mjbots/pi3hat/pull/17*issuecomment-1857817509__;Iw!!K-Hz7m0Vt54!jCtW8IPyUSzprNqWBUXrCEpED6e5eODfFDzJpSJubKqF0FjxgQ49StIzw9GAq7n0bOstchT8yFuJBoNHvzVDOSW-I4U$), or [unsubscribe](https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/ABF62HJMH5N52DQRK4N5SK3YJQ77LAVCNFSM6AAAAABAVPZGPKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNJXHAYTONJQHE__;!!K-Hz7m0Vt54!jCtW8IPyUSzprNqWBUXrCEpED6e5eODfFDzJpSJubKqF0FjxgQ49StIzw9GAq7n0bOstchT8yFuJBoNHvzVDYZpQWLU$).
>> You are receiving this because you authored the thread.Message ID: ***@***.***>

jpieper commented:
Merged in c8b9e13c1a02ece6a9b4dee6acb3aebf182c7c05

jpieper commented:
Thanks a lot!
